src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -38,6 +45,78 @@ public static void ProcessLogFile(string filePath, Solution solution, string too
+
+                if (logText.Contains(@"""version"": ""1.0.0"""))
+                {
+                    // They're opening a v1 log, so we need to transform it
+                    // Ask if they'd like to save the v2 log
+                    int response = VsShellUtilities.ShowMessageBox(SarifViewerPackage.ServiceProvider,
+                                                                   Resources.TransformV1_DialogMessage,
+                                                                   null, // title
+                                                                   OLEMSGICON.OLEMSGICON_QUERY,
+                                                                   OLEMSGBUTTON.OLEMSGBUTTON_YESNOCANCEL,
+                                                                   OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
+
+                    if (response == IDCANCEL)
+                    {
+                        return;
+                    }
+
+                    JsonSerializerSettings settingsV1 = new JsonSerializerSettings()
+                    {
+                        ContractResolver = SarifContractResolverVersionOne.Instance,
+                    };
+
+                    SarifLogVersionOne v1Log = JsonConvert.DeserializeObject<SarifLogVersionOne>(logText, settingsV1);
+                    var transformer = new SarifVersionOneToCurrentVisitor();
+                    transformer.VisitSarifLogVersionOne(v1Log);
+                    log = transformer.SarifLog;
+
+                    if (response == IDYES)
+                    {
+                        // Prompt for a location to save the transformed log
+                        var saveFileDialog = new SaveFileDialog();
+
+                        saveFileDialog.Title = Resources.SaveTransformedV1Log_DialogTitle;
+                        saveFileDialog.Filter = "SARIF log files (*.sarif)|*.sarif";
+                        saveFileDialog.RestoreDirectory = true;
+
+                        filePath = Path.GetFileNameWithoutExtension(filePath) + ".v2.sarif";
+                        saveFileDialog.FileName = Path.GetFileName(filePath);
+                        saveFileDialog.InitialDirectory = Path.GetDirectoryName(filePath);
+
+                        if (saveFileDialog.ShowDialog() != DialogResult.OK)
+                        {
+                            return;
+                        }
+
+                        filePath = saveFileDialog.FileName;
+                        string error = null;
+
+                        try
+                        {
+                            File.WriteAllText(filePath, JsonConvert.SerializeObject(log, settingsV2));
+                        }
+                        catch (UnauthorizedAccessException)
+                        {
+                            error = string.Format(Resources.SaveTransformedV1LogFail_Access_DialogMessage, filePath);
+                        }
+                        catch (IOException ex)
```

Comment:
`Exception` instead of `IOException`?

It used to be that the guidelines said not to catch `Exception`, because there were some exceptions (like `StackOverflowException` and `ThreadAbortException` that were dangerous to catch.

I _think_ that nowadays user code can't catch those dangerous exceptions, and it's ok to catch `Exception`. Please look this up and change if appropriate.

Reply:
This is true (the CLR will automatically re-throw those). Still, I prefer to be explicit, especially with a method that can throw so many different exceptions. I'd like to leave this one.

---
In reply to: [214177624](https://github.com/Microsoft/sarif-visualstudio-extension/pull/23#discussion_r214177624) [](ancestors = 214177624)

Reply:
Could you tell me a little more? As you say, there might be many types of exceptions. As the code stands, most of them will not be caught, and will (presumably cause the extension to crash).  That can't be the desired behavior, right?

Reply:
Definitely not. I'll catch everything rather than handling all of them separately.

---
In reply to: [214191779](https://github.com/Microsoft/sarif-visualstudio-extension/pull/23#discussion_r214191779) [](ancestors = 214191779)

Reply:
Sorry, I wasn't clear. I certainly didn't mean to suggest that you have several catch clauses, one for each type of exception. I meant to suggest that you add a `catch (Exception ex)` to your existing `try/catch` construct.

Do you have a top-level unhandled-exception-catcher already?

Reply:
(I mean, in some other file?)

Reply:
Ah! No wonder I didn't see it. :-) Ok, then, standing by.

Reply:
Sorry, I was confused, because when you wrote "I'd like to leave this one," it sounded like "Won't fix". My mistake.